src/compiler/checker.ts

```diff
@@ -31165,7 +31167,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             && getThisContainer(node, /*includeArrowFunctions*/ true, /*includeClassComputedPropertyName*/ false) === getDeclaringConstructor(prop);
     }
 
-    function checkPropertyAccessExpressionOrQualifiedName(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, leftType: Type, right: Identifier | PrivateIdentifier, checkMode: CheckMode | undefined) {
+    function checkPropertyAccessExpressionOrQualifiedName(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, leftType: Type, right: Identifier | PrivateIdentifier, checkMode: CheckMode | undefined, writeOnly?: boolean) {

```

Comment:
Is the parameter here (and elsewhere) needed anymore now that we're changing `accessKind`?

Reply:
Yes, it's needed in function `checkAssignmentOperatorWorker` to check assignability when the operator is a compound assignment. Without this flag, `checkPropertyAccessExpressionOrQualifiedName` will always use the getter's type when the operator is a compound assignment, as the setter's type is only really needed when checking assignability.